shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/CreateStatisticsService.java

```diff
@@ -42,46 +47,124 @@ private File recreateFile(final String fileName) throws IOException {
+                 ZipOutputStream zos = new ZipOutputStream(fos)) {
+
+                ZipEntry zipEntry = new ZipEntry("shanoirExportStatistics_" + event.getId() + ".tsv");
+
+                zos.putNextEntry(zipEntry);
+                OutputStreamWriter outputStreamWriter = new OutputStreamWriter(zos);
+                BufferedWriter writer = new BufferedWriter(outputStreamWriter);
+                try {
+
+                    while (true) {
+                        event.setMessage("Querying results: " + (startRow + blocSize) + "/" + procedureSize);
+                        eventService.publishEvent(event);
+                        //"getStatistics" is the name of the MySQL procedure
+                        StoredProcedureQuery query = entityManager.createStoredProcedureQuery("getStatistics");
+
+                        //Declare the parameters in the same order
+                        query.registerStoredProcedureParameter(1, String.class, ParameterMode.IN);
+                        query.registerStoredProcedureParameter(2, String.class, ParameterMode.IN);
+                        query.registerStoredProcedureParameter(3, String.class, ParameterMode.IN);
+                        query.registerStoredProcedureParameter(4, String.class, ParameterMode.IN);
+                        query.registerStoredProcedureParameter(5, Integer.class, ParameterMode.IN);
+                        query.registerStoredProcedureParameter(6, Integer.class, ParameterMode.IN);
+
+                        //Pass the parameter values
+                        query.setParameter(1, studyNameInRegExp);
+                        query.setParameter(2, studyNameOutRegExp);
+                        query.setParameter(3, subjectNameInRegExp);
+                        query.setParameter(4, subjectNameOutRegExp);
+                        query.setParameter(5, startRow);
+                        query.setParameter(6, blocSize);
+
+                        //Execute query
+                        @SuppressWarnings("unchecked")
+                        List<Object[]> results = query.getResultList();
+
+                        if (results.isEmpty()) {
+                            break;
+                        }
+
+                        for (Object[] or : results) {
+                            progress += 1f / procedureSize;
+                            event.setProgress(progress);
+                            eventService.publishEvent(event);
+                            List<String> strings = Arrays.stream(or).map(object -> Objects.toString(object, null)).collect(Collectors.toList());
+                            writer.write(String.join("\t", strings));
+                            writer.write('\n');
+                        }
+                        startRow += blocSize;
+                    }
+
+                    writer.flush();
+
+                } finally {
+                    outputStreamWriter.flush();
```

Comment:
Hi @pierrehenri-dauvergne, I wonder if you missed the close here, what might be bad for the memory?